### ISSUE TITLE
Kobo: Transparent widget does not properly refresh after page change via button

### ISSUE THREAD

**Issue by fbriere** [koreader/koreader#12402]
* KOReader version: v2024.07-123
* Device: Kobo Libra (but _not_ Forma, see below)

#### Issue

When a widget that can jump to a different page is made transparent, it does not render properly after jumping to another page via tapping a button.  (In other words, using the skim widget's progress bar does not seem to trigger this.)  So far, I have reproduced this with the skim and fulltext search widgets; let me know if there are others I should try.

The interesting part is that while this (almost) always happens on my Libra, it never does on my Forma.  Seems like it's rather hardware-specific, hence why I assumed it only affects Kobo models.

#### Steps to reproduce

- Open the skim widget
- Make it transparent
- Tap the `+1` button

![transparent2](https://github.com/user-attachments/assets/3a6f48c2-cb97-499d-ba16-c37acd9ee95d)

**Comment by fbriere** [comment]
Here's another example with the fulltext search:

![transparent3](https://github.com/user-attachments/assets/a6b63245-85fe-40f0-b385-50c11562e199)

**Comment by NiLuJe** [comment]
For reference, that's a Libra, first of its name, right?

**Comment by NiLuJe** [comment]
And do you have button flashing *disabled* in the Screen > eInk Settings?

**Comment by poire-z** [comment]
(I'm not experiencing this, with flashing buttons enabled or disabled, on my Kobo GloHD.)

**Comment by NiLuJe** [comment]
Not seeing it on a H2O either, but I *have* seen sunxi devices being confused about it in a somewhat similar fashion, but we've worked around it.

Can't do that on the affected device here, though ;).

**Comment by NiLuJe** [comment]
(The buttons here are also some of the rare ones doing things in a slightly different order in order to show the page before the hiighlights, otherwise, on most devices, you'd get a weird delay with button highlights enabled).

**Comment by NiLuJe** [comment]
> Not seeing it on a H2O either

Nothing on a Libra Colour either, with or without color.

(Not checking on sunxi because, as mentioned earlier, they're special snowflakes ^^).

**Comment by fbriere** [comment]
Yes, it's the OG Libra.  ☺  (Bought within 1-2 months of release, so it's about as original as one can get.)

And while I normally have button flashing enabled, disabling it does indeed seem to resolve the issue.

**Comment by NiLuJe** [comment]
Hmm. So I guess you can try dropping the `vsync` flag from the SkimTo buttons (e.g., https://github.com/koreader/koreader/blob/4910afffddb9ded5d2e8c02f797885b7c2a3f985/frontend/ui/widget/skimtowidget.lua#L126), and that should also work with button flashing enabled?

(I mean, it will *probably* delay the page turn to *after* the flash, too, since that's what this hack is trying to fix ;p).

**Comment by NiLuJe** [comment]
Welp, you might want to user-patch the Button class to ignore the vsync on that device, because that's the first time I've seen it causing issues (and in the vast majority of cases, it actually *helps*, I swear ;p).

The dirty but efficient approach is just to copy the full `Button:onTapSelectButton` method without the vsync branches.
The nasty but future-proof approach is to wrap `Button:init` to always set `self.vsync` to false ;).

**Comment by NiLuJe** [comment]
> The nasty but future-proof approach is to wrap `Button:init` to always set `self.vsync` to false ;).

e.g., 

`2-no-button-vsync.lua`
```lua
local Button = require("ui/widget/button")

local orig_init = Button.init
function Button:init()
    orig_init(self)
    self.vsync = false
end
```
